polaris-react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -114,7 +114,7 @@ export function Tooltip({
   const hoverOutTimeout = useRef<NodeJS.Timeout | null>(null);
 
   const handleFocus = useCallback(() => {
-    if (originalActive !== false) {
+    if (Boolean(originalActive) !== false) {
```

Comment:
If `originalActive` is `undefined`, `originalActive !== false` evaluates to true, so the `setActiveTrue()` fires and sets the local `active` state to true.